runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -868,6 +868,14 @@ void chpl_comm_init(int *argc_p, char ***argv_p) {
   setup_ibv();
   setup_polling();
 
+  // PSHM needs an external progress thread to guarantee AM handlers make
+  // progress in the absence of other GASNet calls, so if we've disabled our
+  // external progress thread then we should disable PSHM
+  if (!pollingRequired) {
+    // disable PSHM even if it was enabled during configuration.
+    chpl_env_set("GASNET_SUPERNODE_MAXSIZE", "1", 0);
```

Comment:
Should this also show a warning telling the user what happened so they are not suprised?

Reply:
I don't think so. It's not like the user requested PSHM and we are ignoring their request, it's that by default the `gasnet` library is now configured with PSHM enabled but we have to disable it at runtime in the `gasnet` shim because that shim doesn't create an external polling thread. Previously, PSHM was not enabled during configuration so this wasn't an issue.